### PR TITLE
init load

### PR DESCRIPTION
**Pull request by paulr34**

### CODE REVIEW

File: src/pages/ZapConfig.vue [hunk 1]
```diff
@@ -311,17 +311,15 @@ export default {
           this.selectedZclGenData.length != 0
         ) {
           let data = {
```
Comment (paulr34) [inline src/pages/ZapConfig.vue:313]: This file has this code: 
`  beforeCreate() {
    // setCssVar('primary', '#33F')
    this.$serverGet(restApi.uri.initialPackagesSessions).then((result) => {
      this.zclPropertiesRow = result.data.zclProperties
      this.selectedZclPropertiesData = result.data.zclProperties[0]
      this.zclGenRow = result.data.zclGenTemplates`

Which is the first REST API call. If session creation is in http-server then the session would be created by this REST API, I believe.

File: src-electron/rest/file-ops.js [hunk 2]
```diff
@@ -41,24 +41,31 @@ const studio = require('../ide-integration/studio-rest-api')
...
-          // store studio project path
-          await querySession.updateSessionKeyValue(
-            db,
-            req.zapSessionId,
-            dbEnum.sessionKey.ideProjectPath,
-            ideProjectPath
-          )
-        }
+        //  if (ideProjectPath) {
+        //env.logInfo(`IDE: setting project path(${name}) to ${ideProjectPath}`)
```
Comment (jingteng25742) [inline src-electron/rest/file-ops.js:61]: ideProjectPath is here so ZAP knows how to interact with the Studio IDE.
The logInfo should stay to make debugging possible.

File: src-electron/rest/file-ops.js [hunk 3]
```diff
@@ -41,24 +41,31 @@ const studio = require('../ide-integration/studio-rest-api')
 function httpPostFileOpen(db) {
   return async (req, res) => {
     let { zapFilePath, ideProjectPath } = req.body
-    let name = ''
+    if (req.body.search?.includes('filePath=%2F')) {
```
Comment (jingteng25742) [inline src-electron/rest/file-ops.js:44]: The ideProjectPath is passed in from the Studio side already. 
Is there any specific reason why it needs to be embedded into "search" and why is it called "search"?

File: src-electron/rest/file-ops.js [hunk 4]
```diff
@@ -41,24 +41,31 @@ const studio = require('../ide-integration/studio-rest-api')
...
+      ideProjectPath = dbEnum.sessionKey.ideProjectPath
+      zapFilePath = req.body.search?.split('filePath=')
```
Comment (jingteng25742) [inline src-electron/rest/file-ops.js:46]: I see the "search" variablle is being reparsing the string again here.
Why isn't the content of the "filePath" passed as "zapFilePath" instead?

**Review by jingteng25742 — CHANGES_REQUESTED** [review]

**Comment by tecimovic** [comment]
@paulr34 : can you please rebase this branch?

**Comment by paulr34** [comment]
> @paulr34 : can you please rebase this branch?

done

File: src-electron/rest/file-ops.js (hunk 3 above)
Comment (paulr34) [reply src-electron/rest/file-ops.js:44]: search is an array object within "Window.Location" which was already storing objects. I just added a new one to conform with what was already there.

File: src-electron/ui/window.ts [hunk 5]
```diff
@@ -130,16 +134,13 @@ export function windowCreate(port: number, args?: WindowCreateArgs) {
     args?.uiMode,
     args?.standalone,
     args?.new,
+    args?.filePath,
```
Comment (paulr34) [inline src-electron/ui/window.ts:137]: @jingteng25742 this variable gets added to the "search" object array

File: src-electron/rest/file-ops.js (hunk 2 above)
Comment (paulr34) [reply src-electron/rest/file-ops.js:69]: agree, will fix - thanks

File: src-electron/rest/file-ops.js (hunk 3 above)
Comment (paulr34) [reply src-electron/rest/file-ops.js:44]: to further explain, I completely changed how ZAP opens a file. It no longer uses the API renderer.

File: src-electron/rest/file-ops.js (hunk 4 above)
Comment (paulr34) [reply src-electron/rest/file-ops.js:46]: had to redo how the rest API which opens files is called

File: cypress/integration/preview_button/preview_button.spec.js [hunk 6]
```diff
@@ -1,29 +0,0 @@
-/// <reference types="cypress" />
```
Comment (brdandu) [inline cypress/integration/preview_button/preview_button.spec.js:1]: Why is this removed?

File: cypress/fixtures/baseurl.json [hunk 7]
```diff
@@ -1,3 +1,3 @@
 {
-  "baseurl": "http://localhost:8080/?restPort=9070"
+  "baseurl": "http://localhost:9070"
```
Comment (brdandu) [inline cypress/fixtures/baseurl.json:2]: Why was this changed?

File: src-electron/db/db-mapping.js [hunk 8]
```diff
@@ -610,7 +610,7 @@ exports.map = {
       sessionId: x.SESSION_ID,
       sessionKey: x.SESSION_KEY,
       creationTime: x.CREATION_TIME,
-      packageRef: x.PACKAGE_REF,
```
Comment (brdandu) [inline src-electron/db/db-mapping.js:613]: Can this cause backwards compatibility issues? We should not be changing the mapping in general. Keep the existing ones for backwards compatibility.

File: src-electron/importexport/import-json.js [hunk 9]
```diff
@@ -380,7 +380,7 @@ async function importEndpointTypes(
  */
 async function jsonDataLoader(db, state, sessionId, packageMatch) {
   // Initially clean up all the packages from the session.
-  await queryPackage.deleteAllSessionPackages(db, sessionId)
+  // await queryPackage.deleteAllSessionPackages(db, sessionId)
```
Comment (brdandu) [inline src-electron/importexport/import-json.js:383]: Cleanup

File: src-electron/rest/file-ops.js [hunk 10]
```diff
@@ -41,8 +41,17 @@ const studio = require('../ide-integration/studio-rest-api')
...
+      zapFilePath = zapFilePath[1]?.replaceAll('%2F', '//').trim()
+      console.log('NOTICE ME!!!')
+      console.log(zapFilePath)
```
Comment (brdandu) [inline src-electron/rest/file-ops.js:49]: Get rid of console.log messages.

File: src-electron/rest/file-ops.js [hunk 11]
```diff
@@ -41,8 +41,17 @@ const studio = require('../ide-integration/studio-rest-api')
...
+    }
 
+    // let { ideProjectPath } = req.body
+    let name = ''
+    //console.log(zapFilePath)
```
Comment (brdandu) [inline src-electron/rest/file-ops.js:54]: Get rid of consoleless.log messages

File: src-electron/rest/file-ops.js [hunk 12]
```diff
@@ -41,8 +41,17 @@ const studio = require('../ide-integration/studio-rest-api')
...
-    let name = ''
+    if (req.body.search?.includes('filePath=%2F')) {
+      ideProjectPath = dbEnum.sessionKey.ideProjectPath
+      zapFilePath = req.body.search?.split('filePath=')
+      zapFilePath = zapFilePath[1]?.replaceAll('%2F', '//').trim()
+      console.log('NOTICE ME!!!')
+      console.log(zapFilePath)
+    }
 
+    // let { ideProjectPath } = req.body
```
Comment (brdandu) [inline src-electron/rest/file-ops.js:52]: cleanup

File: src-electron/rest/initialize.js [hunk 13]
```diff
@@ -58,20 +166,14 @@ function packagesAndSessions(db) {
...
-    if (!req.body.newConfiguration) {
-      await util.ensurePackagesAndPopulateSessionOptions(
-        db,
-        sessionId,
-        {},
-        req.body.zclProperties,
-        req.body.genTemplate
-      )
-    }
+    console.log(req.body.sessionId)
```
Comment (brdandu) [inline src-electron/rest/initialize.js:169]: get rid of console.log message

File: src-electron/util/util.js [hunk 14]
```diff
@@ -114,6 +116,8 @@ async function ensurePackagesAndPopulateSessionOptions(
           )
         }
         if (packageId != null) {
+          console.log('PACKAGEID-zcl')
+          console.log(packageId)
```
Comment (brdandu) [inline src-electron/util/util.js:120]: get rid of console.log message

File: src-electron/util/util.js [hunk 15]
```diff
@@ -164,6 +168,8 @@ async function ensurePackagesAndPopulateSessionOptions(
               }
             }
             if (packageId != null) {
+              console.log('PACKAGEID-gen')
+              console.log(packageId)
```
Comment (brdandu) [inline src-electron/util/util.js:172]: get rid of console.log message

File: src/api/renderer_api.js [hunk 16]
```diff
@@ -47,17 +47,6 @@ export function renderer_api_info() {
   }
 }
 
-function fnOpen(zapFilePath, ideProjectPath) {
```
Comment (brdandu) [inline src/api/renderer_api.js:50]: is this not needed by Studio?

**Review by brdandu — CHANGES_REQUESTED** [review]

File: src/api/renderer_api.js (hunk 16 above)
Comment (paulr34) [reply src/api/renderer_api.js:50]: the whole point is to get rid of this

File: src-electron/rest/initialize.js [hunk 17]
```diff
@@ -25,27 +25,135 @@ const querySession = require('../db/query-session.js')
...
-    )
-    const zclGenTemplates = await queryPackage.getPackagesByType(
-      db,
-      dbEnum.packageType.genTemplatesJson
-    )
-    const sessions = await querySession.getDirtySessionsWithPackages(db)
+    if (req.body.search?.includes('filePath=%2F')) {
+      let filePath = req.body.search.split('filePath=')
+      filePath = filePath[1].replaceAll('%2F', '//').trim()
+      console.log(filePath)
```
Comment (brdandu) [inline src-electron/rest/initialize.js:41]: get rid of console.log message

File: src-electron/rest/initialize.js [hunk 18]
```diff
@@ -25,27 +25,135 @@ const querySession = require('../db/query-session.js')
...
+          dbEnum.packageType.zclProperties,
+          category
+        )
+        console.log(dbEnum.packageType.zclProperties)
+        const zclGenTemplates = await queryPackage.getPackagesByCategoryAndType(
+          db,
+          dbEnum.packageType.genTemplatesJson,
+          category
+        )
+        console.log(zclGenTemplates)
```
Comment (brdandu) [inline src-electron/rest/initialize.js:59]: get rid of console.log message

File: src-electron/rest/initialize.js [hunk 19]
```diff
@@ -25,27 +25,135 @@ const querySession = require('../db/query-session.js')
...
+        const sessions = await querySession.getDirtySessionsWithPackages(db)
+        console.log(sessions)
```
Comment (brdandu) [inline src-electron/rest/initialize.js:61]: get rid of console.log message

File: src-electron/rest/initialize.js [hunk 20]
```diff
@@ -25,27 +25,135 @@ const querySession = require('../db/query-session.js')
...
+}
+
+function sessionCreate(db) {
+  return async (req, res) => {
+    let { zclProperties, genTemplate } = req.body
+    let sessionUuid = req.query[restApi.param.sessionId]
+    let userKey = req.session.id
+
+    if (sessionUuid == null || userKey == null) {
+      console.log('empty')
```
Comment (brdandu) [inline src-electron/rest/initialize.js:115]: get rid of console.log message

**Review by brdandu — CHANGES_REQUESTED** [review]

**Review by tecimovic — APPROVED** [review]
Good to go, fix the small cleanup in a separate PR after this big merge.

File: src-electron/util/util.js [hunk 21]
```diff
@@ -61,18 +61,16 @@ function checksum(data) {
 async function ensurePackagesAndPopulateSessionOptions(
   db,
   sessionId,
-  options,
-  selectedZclPropertyPackage = null,
+  options = null,
+  selectedZclPropertyPackage = [],
```
[the remark this answers is not in the document]
Comment (paulr34) [reply src-electron/util/util.js:65]: will fix, didn't know that

File: src-electron/rest/file-ops.js [hunk 22]
```diff
@@ -40,25 +40,28 @@ const studio = require('../ide-integration/studio-rest-api')
...
-          env.logInfo(`IDE: setting project path(${name}) to ${ideProjectPath}`)
-          // store studio project path
-          await querySession.updateSessionKeyValue(
-            db,
-            req.zapSessionId,
-            dbEnum.sessionKey.ideProjectPath,
-            ideProjectPath
-          )
-        }
+        //  if (ideProjectPath) {
```
[the remark this answers is not in the document]
Comment (paulr34) [reply src-electron/rest/file-ops.js:64]: okay will make a ticket and fix

File: src-electron/db/query-session.js [hunk 23]
```diff
@@ -163,11 +163,7 @@ async function getSessionFromSessionId(db, sessionId) {
  */
 async function getSessionInfoFromSessionKey(db, sessionKey) {
   return dbApi
-    .dbGet(
-      db,
-      'SELECT SESSION_ID, SESSION_KEY, CREATION_TIME FROM SESSION WHERE SESSION_KEY = ?',
-      [sessionKey]
-    )
+    .dbGet(db, 'SELECT * FROM SESSION WHERE SESSION_KEY = ?', [sessionKey])
```
[the remark this answers is not in the document]
Comment (paulr34) [reply src-electron/db/query-session.js:166]: ill figure out a better way to parse that URL and okay I will select all of the items specifically. I'll add this to the next PR work. Thank you.

File: src-electron/importexport/import-json.js (hunk 9 above)
[the remark this answers is not in the document]
Comment (paulr34) [reply src-electron/importexport/import-json.js:383]: okay will delete, thank you

File: src-electron/importexport/import.js [hunk 24]
```diff
@@ -95,6 +96,14 @@ async function importDataFromFile(
     let sid
     if (options.sessionId == null) {
       sid = await querySession.createBlankSession(db)
+      await util.ensurePackagesAndPopulateSessionOptions(
+        db,
+        sid,
+        {
+          zcl: env.builtinSilabsZclMetafile(),
+          template: env.builtinTemplateMetafile(),
+        }, null, null
```
[the remark this answers is not in the document]
Comment (paulr34) [reply src-electron/importexport/import.js:105]: I doubt it, probably needs to be cleaned up, thanks

**Review by jingteng25742 — CHANGES_REQUESTED** [review]

**Review by brdandu — APPROVED** [review]

File: src-electron/rest/file-ops.js (hunk 3 above)
Comment (paulr34) [reply src-electron/rest/file-ops.js:44]: @tecimovic the "search" object has many different parameters. In order to find the file I had to parse the entire string. @jingteng25742 found a better way to do it which I will make a new PR and fix.

**Review by jingteng25742 — APPROVED** [review]
